<launch>
<!-- ============ Arguments ============ -->
  <!-- Groundtruth -->
  <arg name="ground_truth_pointcloud_file" default="/home/lukas/Documents/Datasets/flat_dataset/ground_truth/run2/flat_2_gt_10000.ply"/>  <!-- Used if not using rio. -->
  <arg name="use_rio" default="false"/>   <!-- Otherwise look up scene hash. -->
  <arg name="scene_id" default="0"/>
  <arg name="scan_id" default="0"/>

  <!-- Source directory -->
  <arg name="map_file" default="/home/lukas/Documents/PanopticMapping"/>
  <arg name="output_suffix" default=""/>

  <!-- Evaluation Params -->
  <arg name="maximum_distance" default="10000"/>
  <arg name="inlier_distance" default="0.05"/>
  <arg name="ignore_truncated_points" default="false"/>
  <arg name="is_single_tsdf" default="false"/>

<!-- ============ Evaluations ============ -->
  <node name="multi_map_evaluation" pkg="panoptic_mapping_utils" type="multi_map_evaluation" output="screen" required="true">
    <param name="map_file" value="$(arg map_file)" />
    <param name="ground_truth_pointcloud_file" value="$(arg ground_truth_pointcloud_file)" unless="$(arg use_rio)"/>
    <param name="output_suffix" value="$(arg output_suffix)" />
    <param name="maximum_distance" value="$(arg maximum_distance)" />
    <param name="visualize" value="false" />
    <param name="evaluate" value="true" />
    <param name="compute_coloring" value="false" />
    <param name="inlier_distance" value="$(arg inlier_distance)" />
    <param name="ignore_truncated_points" value="$(arg ignore_truncated_points)" />
    <param name="is_single_tsdf" value="$(arg is_single_tsdf)" />
  </node>

  <node name="evaluation_manager" pkg="panoptic_mapping_utils" type="evaluation_manager.py" output="screen" required="true">
    <param name="map_file" value="$(arg map_file)" />
    <param name="use_rio" value="$(arg use_rio)" />
    <param name="scene_id" value="$(arg scene_id)" />
    <param name="scan_id" value="$(arg scan_id)" />
  </node>
</launch>
